src/appleseed/foundation/utility/autoreleaseptr.h

```diff
@@ -147,6 +147,11 @@ class auto_release_ptr
         return m_ptr;
     }
 
+    operator bool() const
```

Comment:
Looks good! Can you add a comment please? Something like

    // Implicit conversion of the wrapped pointer to a bool.

Reply:
Missing explicit, maybe?

Reply:
Comment and explicit added in 2 separate commit so if the explicit wasnt necessary, we just have to revert the last commit.
@est77, why do you think explicit is important for this function?

Reply:
@est77 Good catch, I hadn't noticed that the conversion was explicit on `std::unique_ptr<>`.

@Uedaki Ideally we would like the conversion to be explicit to mimic `std::unique_ptr<>` and other standard smart pointers. The reason they're marked as explicit is to avoid things like

    std::unique_ptr<int> x = std::make_unique(42);
    const bool b = x;  // error

while allowing

    std::unique_ptr<int> x = std::make_unique(42);
    if (x)  // OK
    {
        ...
    }

Unfortunately, as you may have noticed, explicit conversion operators aren't supported by VS 2012 which we still need to support. I suggest making explicit conditional to allow catching errors on newer compilers:

        // Bool conversion of the wrapped pointer.
    #if defined _MSC_VER && _MSC_VER >= 1800
        explicit
    #endif
        operator bool() const
        {
            return static_cast<bool>(m_ptr);
        }

Make sure to remove the word "implicit" from the comment as it's no longer the case.